test/applications_integration_test.go

```diff
@@ -846,6 +856,132 @@ func theContentsOfTheBoxWithNameShouldBeIfThereIsAnErrorItIs(encodedBoxName, box
 	return nil
 }
 
+func currentApplicationShouldHaveFollowingBoxes(fromClient, encodedBoxesRaw string) error {
+	var expectedNames [][]byte
+	if len(encodedBoxesRaw) > 0 {
+		encodedBoxes := strings.Split(encodedBoxesRaw, ":")
+		expectedNames = make([][]byte, len(encodedBoxes))
+		for i, b := range encodedBoxes {
+			expected, err := base64.StdEncoding.DecodeString(b)
+			if err != nil {
+				return err
+			}
+			expectedNames[i] = expected
+		}
+	}
+
+	var r models.BoxesResponse
+	var err error
+
+	if fromClient == "algod" {
+		r, err = algodV2client.GetApplicationBoxes(applicationId).Do(context.Background())
+	} else if fromClient == "indexer" {
+		r, err = indexerV2client.SearchForApplicationBoxes(applicationId).Do(context.Background())
+	} else {
+		err = fmt.Errorf("expecting algod or indexer, got " + fromClient)
+	}
+	if err != nil {
+		return err
+	}
+
+	actualNames := make([][]byte, len(r.Boxes))
+	for i, b := range r.Boxes {
+		actualNames[i] = b.Name
+	}
+
+	if len(expectedNames) != len(actualNames) {
+		return fmt.Errorf("expected and actual box names length do not match:  %v != %v", len(expectedNames), len(actualNames))
+	}
+
+	contains := func(elem []byte, xs [][]byte) bool {
+		for _, x := range xs {
+			if bytes.Equal(elem, x) {
+				return true
+			}
+		}
+		return false
+	}
+
+	for _, e := range expectedNames {
+		if !contains(e, actualNames) {
+			return fmt.Errorf("expected and actual box names do not match: %v != %v", expectedNames, actualNames)
+		}
+	}
+
+	return nil
+}
+
+func sleptForNMilliSecsForIndexer(n int) error {
+	time.Sleep(time.Duration(n) * time.Millisecond)
+	return nil
+}
+
+func currentApplicationShouldHaveBoxNum(fromClient string, max int, expectedNum int) error {
+	var r models.BoxesResponse
+	var err error
+
+	if fromClient == "algod" {
+		r, err = algodV2client.GetApplicationBoxes(applicationId).Max(uint64(max)).Do(context.Background())
+	} else if fromClient == "indexer" {
+		r, err = indexerV2client.SearchForApplicationBoxes(applicationId).Limit(uint64(max)).Do(context.Background())
+	} else {
+		err = fmt.Errorf("expecting algod or indexer, got " + fromClient)
+	}
+	if err != nil {
+		return err
+	}
+
+	if len(r.Boxes) != expectedNum {
+		return fmt.Errorf("expected and actual box number do not match: %v != %v", expectedNum, len(r.Boxes))
+	}
+	return nil
+}
+
+func indexerSaysCurrentAppShouldHaveTheseBoxes(max int, next string, encodedBoxesRaw string) error {
```

Comment:
Is this step still needed? I see that `currentApplicationShouldHaveFollowingBoxes` has logic for indexer as well as algod.

Reply:
Yes it is because of the `next` parameter.

Resolving